src/DynamoCoreWpf/UI/Converters.cs

```diff
@@ -2660,5 +2630,89 @@ public object ConvertBack(object value, Type targetType, object parameter, Syste
                 throw new Exception("The method or operation is not implemented.");
             }
 
-        }               
-   }
+        }
+
+        public class ClassViewMarginConverter : IValueConverter
+        {
+            private const int levelmargin = 45;
+            private const int marginTop = -10;
+            private const int marginLeft = -10;
+            private const int viewMarginLeft = -30;
+            private const int factor = 10;
+
+            public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+            {
+                var grid = value as Grid;
+                
+                //Outerbuttongrid has two different child.
+                if (grid != null && grid.Children.Count > 0)
+                {
+                    //first Child is a border
+                    var child1 = grid.Children[0] as Border;
+                    if (child1 != null)
```

Comment:
Don't build deeply nested if blocks, just do:

```
if (child1 == null) return x;
```

This is real tough to read.